### ISSUE TITLE
Legal?

### ISSUE THREAD

**Issue by seeeeew** [z0z0z/mf-install#1]
Is it legal to distribute these dll files? Where do they come from?

**Comment by z0z0z** [comment]
These are freely available for download from Microsoft and many legal third party party DLL download sites, which Microsoft has not bothered.

This is solely for personal and noncommercial use. I don't see there being a legal issue here, but if you happen know a Microsoft lawyer I'd listen to them, as that seems to be the only person in the world who might have a true answer on this.

**Comment by TiZ-EX1** [comment]
This repo was posted as a workaround for SoulCalibur VI in its issue on the proton tracker, but the link was removed because kisak said it was legally problematic. (not going to reference the issue number or create a link because doing so would create a link there too)

Microsoft not bothering a site doesn't make it legal, and I'd definitely trust a Valve employee on this. If these dlls are freely available from Microsoft under terms that don't conflict with using them for this purpose, then they should be downloaded by the script the same way that winetricks does its scripts.

**Comment by z0z0z** [comment]
@TiZ-EX1It's in a legal gray area. It's fully understandable that Valve wouldn't allow links to this on their issue tracker, but I'd eat my hat if Microsoft cares, because they definitely don't, and are trying very hard not to look like the bad guy anymore.

As far as I know, no one has found MS servers with these files available.

**Comment by telans** [comment]
The dlls are availlable in [here](https://www.microsoft.com/en-us/software-download/mediafeaturepack) when extracted with something like p7zip, but the generated links are only available for ~24 hours. Windows 10 too

**Comment by TiZ-EX1** [comment]
"MS doesn't care / is trying not to look like a bad guy" is not how usage rights work, but it seems you have a well enough understanding of why Valve can't endorse this. I believe the Wine developers are trying to implement Media Foundation, which would be the "right" fix.

Winetricks currently pulls mf.dll--but none of the other dlls in this repo--from an update package. I know mf-installcab pulls some other files from update packages, but many of the files here are still unaccounted for and they're probably the main reason it works.

I am currently using this fix, because it does get SoulCalibur's title screen to work, and probably also the reported crash from trying to start the main story mode. (Haven't tried to start the main story yet.) Would you be interested in taking a pull request for a corresponding uninstall script for anyone who wants to go back to a kosher prefix state?

**Comment by z0z0z** [comment]
@TiZ-EX1 Lately Microsoft has been trying very hard to embrace their open source and Linux friends. Big daddy Microsoft ain't evil no mo they love Linux and mf-install.

Anyway regarding your pull request question, replacing the old DLL files (which are mostly just stub placeholders) and turning off the native overrides wouldn't be too hard, not sure about reverting the registry entries though.

I've never had a problem just making new prefixes instead of trying to remove this personally.

It would depend what it looks like.

**Comment by TiZ-EX1** [comment]
> ain't evil no mo [...] love *mf-install*

I'm going to pretend that you're joking and not delusional.

Considering the registry sticking point, creating a new prefix--or backing up any prefix you want to run the fix on--probably is indeed easier so I won't spend the effort.

**Comment by jp-bennett** [comment]
Are the needed DLLs available on the freely downloadable Windows 10 ISOs? A silly but technically legal workaround might be to download said ISO and extract the dlls on the end machine.

**Comment by telans** [comment]
They are, you can extract them from the various cab files within some isos

**Comment by z0z0z** [comment]
The DLL files are from Windows 7 (and need to be from Windows 7).

Win10 ISOs are easily available for download, it's a large file, but a script could just download the ISO and extract them from it.

However, the Win7 ISOs are no longer freely available for download from Microsoft, they require a Win7 key to generate a download link. They obviously only do this to discourage Windows 7 use.

There's no easy legal way for a script to download these files from Microsoft.

Crappy "DLL download" sites have operated since forever, distributing actual DLL files from Windows, and Microsoft has never bothered them. 

Windows 7 ISOs are actually quite hard to find now, if this script required users to do that, thousands of people who are now playing games on Linux they couldn't before, wouldn't be doing that.

Valve or Glorious Eggroll can't host this, but I certainly will and not care.

**Comment by stuartpb** [comment]
Why do the files need to be from Windows 7? I was having some issues (I think performance-wise, long hitches at the start of videos), so I copied some of my files from my installation of Windows 10, and it still worked fine (and I think may have fixed the issue I was having).

**Comment by z0z0z** [comment]
Some of the DLLs could always be win10.

At least in the past, only Win7 DLLs worked. Maybe that's different now.

**Comment by frostworx** [comment]
Hi,
I'm currently [working on](https://github.com/frostworx/steamtinkerlaunch/issues/32) a seamless MF integration in my [steamtinkerlaunch (stl)](https://github.com/frostworx/steamtinkerlaunch)
While also writing on automatic config [import/convert]((https://github.com/frostworx/steamtinkerlaunch/issues/40)) from [lutris](https://github.com/lutris/lutris/issues/3117) I just "finished" a rewrite of the lutris mf functions (extracting dlls and registry keys from windows6.1-KB976932-X64.exe which is reused from the winetricks cache _(or downloaded into the winetricks cache)_ and one dll directly from lutris) just to find out that some more dlls are still missing which are inside mf-install.
Would it be legal to provide a download option in stl, which would clone mf-install and automatically use it or should I better just use a manual download when found in a specific stl download directory?
Any hints are welcome!

**Comment by z0z0z** [comment]
TLDR; It's technically not legal to redistribute these Microsoft DLLs but no one cares including Microsoft

There's countless crappy DLL download websites that do the same thing.

**Comment by frostworx** [comment]
Thanks for the reply.
I think so as well. In fact I already implemented mf-install directly after my question like described [here](https://github.com/frostworx/steamtinkerlaunch/issues/32#issuecomment-685005350) _(not commited yet)_
So it is completely optionally.
